### PR TITLE
refactor: improved validation for dns.setHosts API endpoint

### PR DESCRIPTION
**Pull request by ifree92**
Added better input validation for `dns.setHosts` endpoint:

- required only one MXE record and only for EmailType=MXE
- required at least one MX record with EmailType=MX
- forbidden MX/MXE records if EmailType != MX/MXE
- required address protocol prefix for record types URL/URL301/FRAME
- required address protocol prefix for CAA iodef record type

### CODE REVIEW

**Review by StyleT — APPROVED** [review]

**Comment by ifree92** [comment]
@adamdecaf sorry for merging before your approval.
Now, this PR is a blocker for me, would like to merge right now, to prepare for v2 2.0.0 public release that is being used by a new coming terraform provider.

If you have any comments - let me know. Thank you.

**Comment by adamdecaf** [comment]
No worries!

**Pull request merged**
